lowdb.d.ts

```diff
@@ -1,530 +1,532 @@
-  ): LoDashWrapper;
-
-  /**
-   * @see _.size
-   */
-  size(): LoDashWrapper;
-
-  /**
-   * @see _.take
-   */
-  take(n?: number): LoDashWrapper;
-
-  /**
-   * @see _.times
-   */
-  times<TResult>(
-    iteratee: (num: number) => TResult
-  ): LoDashWrapper;
-
-  /**
-   * @see _.times
-   */
-  times(): LoDashWrapper;
-
-  /**
-   * @see _.uniqueId
-   */
-  uniqueId(): LoDashWrapper;
-
-  value<T>(): T;
-
-  pop<T>(): T;
-  push<T>(...items: T[]): LoDashWrapper;
-  shift<T>(): T;
-  sort<T>(compareFn?: (a: T, b: T) => number): LoDashWrapper;
-  splice<T>(start: number): LoDashWrapper;
-  splice<T>(start: number, deleteCount: number, ...items: any[]): LoDashWrapper;
-  unshift<T>(...items: T[]): LoDashWrapper;
-}
-
-interface Storage {
-  /**
-    * Reads the database.
-    *
-    * @param source The source location.
-    * @param deserialize The deserialize function to apply.
-    * @return Returns a promise with the deserialized db object.
-  */
-  read?(source: string, deserialize: any): PromiseLike<any>
-  /**
-    * Reads the database.
-    *
-    * @param source The source location.
-    * @param deserialize The deserialize function to apply.
-    * @return Returns the deserialized db object.
-  */
-  read?(source: string, deserialize: any): Object
-  /**
-    * Writes to the database.
-    *
-    * @param destination The destination location.
-    * @param obj The object to write.
-    * @param serialize The serialize function to apply.
-  */
-  write?(destination: string, obj: any, serialize: any): void
-  /**
-    * Writes to the database.
-    *
-    * @param destination The destination location.
-    * @param obj The object to write.
-    * @param serialize The serialize function to apply.
-  */
-  write?(destination: string, obj: any, serialize: any): PromiseLike<void>
+    sortBy<T, TSort>(
+      iteratee?: ListIterator<T, TSort>
+    ): LoDashWrapper;
+
+    /**
+     * @see _.sortBy
+     */
+    sortBy(iteratee: string): LoDashWrapper;
+
+    /**
+     * @see _.sortBy
+     */
+    sortBy<W extends {}>(whereValue: W): LoDashWrapper;
+
+    /**
+     * @see _.sortBy
+     */
+    sortBy(): LoDashWrapper;
+
+    /**
+     * @see _.sortBy
+     */
+    sortBy<T>(...iteratees: (ListIterator<T, boolean> | Object | string)[]): LoDashWrapper;
+
+    /**
+     * @see _.sortBy
+     */
+    sortBy<T>(iteratees: (ListIterator<T, any> | string | Object)[]): LoDashWrapper;
+
+    /**
+     * @see _.slice
+     */
+    slice(
+      start?: number,
+      end?: number
+    ): LoDashWrapper;
+
+    /**
+     * @see _.size
+     */
+    size(): LoDashWrapper;
+
+    /**
+     * @see _.take
+     */
+    take(n?: number): LoDashWrapper;
+
+    /**
+     * @see _.times
+     */
+    times<TResult>(
+      iteratee: (num: number) => TResult
+    ): LoDashWrapper;
+
+    /**
+     * @see _.times
+     */
+    times(): LoDashWrapper;
+
+    /**
+     * @see _.uniqueId
+     */
+    uniqueId(): LoDashWrapper;
+
+    value<T>(): T;
```

Comment:
I feel like this function may return `T | undefined`.

For example:

```
const db = new lowdb(...);
const valid = db.get('users').find({ email: "real-email@email.com" }).value<User>();
const invalid = db.get('users').find({ email: "fake-email@email.com" }).value<User>();
```

The `invalid` will show as `User` even though at runtime it will actually be `undefined`. It kind of blows up the type safety.

Of course, my work-around currently is to do: `...value<User | undefined>();`. Which gets me where I want to be. Just thought I'd point that out.